shared/js/l10n.js

```diff
@@ -1607,20 +1626,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this));
+    } else {
+      registerLocales.call(this);
     }
     return initLocale.call(this);
   }
 
+  function registerLocales(extraLangs) {
```

Comment:
Maybe make `registerLocales` take `availableLanguages` and `defaultLanguage` as arguments?